service-runner.js

```diff
@@ -202,7 +256,15 @@ ServiceRunner.prototype._runMaster = function() {
     process.on('SIGTERM', shutdown_master);
 
     // Set up rolling restarts
-    process.on('SIGHUP', this._rollingRestart.bind(this));
+    process.on('SIGHUP', function() {
```

Comment:
Hm, actually this PR breaks the workflow when `self.options.useDocker === true` completely. The check is done in `processConfig()` on L75, but then the code path proceeds to execute `runMaster()` regardless of this. Previously, the promise chain was broken.